src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/RazorCompletionEndpoint.cs

```diff
@@ -247,26 +229,20 @@ internal static bool IsApplicableTriggerContext(CompletionContext context)
                             Kind = CompletionItemKind.Struct,
                         };
 
-                        if (razorCommitCharacters.Length > 0)
-                        {
-                            directiveCompletionItem = directiveCompletionItem with { CommitCharacters = razorCommitCharacters };
-                        }
+                        directiveCompletionItem.UseCommitCharactersFrom(razorCompletionItem, clientCapabilities);
```

Comment:
We utilize a `UseCommitCharactersFrom` because depending on the capabilities we'll either set the `VsCommitCharacters` or normal `CommitCharacters`. This became super repetitive so built an extension method to make it easier